tests/ut/plugin_demo/customIdx/customIdx.cc

```diff
@@ -0,0 +1,503 @@
+// Copyright (C) 2019-2023 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License.
+
+#include <queue>
+
+#include "customIdx_config.h"
+#include "knowhere/index.h"
+#include "knowhere/index_node.h"
+#include "knowhere/log.h"
+
+namespace demo {
```

Comment:
this file has to contain a loooooooooooooooooooooooooooooooooot of comments, which describes what every piece does